### ISSUE TITLE
Backtrace prints many <unknown function>

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#25714]
Our backtrace code frequently prints out frames with `<unknown function>`, indicating that `backtrace_symbols` was unable to retrieve the symbol for the backtrace. For example:

```
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7efd7464fc11 in /data/users/ezyang/pytorch-tmp/torch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7efd7464f82a in /data/users/ezyang/pytorch-tmp/torch/lib/libc10.so)
frame #2: torch::autograd::VariableType::checked_cast_variable(at::Tensor&, char const*, int) + 0x138 (0x7efd7a777188 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch.so)
frame #3: torch::autograd::VariableType::unpack(at::Tensor&, char const*, int) + 0x9 (0x7efd7a777459 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch.so) 
frame #4: torch::autograd::VariableType::copy_(at::Tensor&, at::Tensor const&, bool) + 0xb0 (0x7efd7a778ff0 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch.so)
frame #5: <unknown function> + 0x6a60c9 (0x7efd970910c9 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch_python.so)
frame #6: <unknown function> + 0x6a7ee9 (0x7efd97092ee9 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch_python.so)
frame #7: c10d::ProcessGroupGloo::runLoop(int) + 0x169 (0x7efd97086199 in /data/users/ezyang/pytorch-tmp/torch/lib/libtorch_python.so)
frame #8: <unknown function> + 0xb8678 (0x7efd96962678 in /home/ezyang/Dev/pytorch-tmp-env/lib/libstdc++.so.6)
frame #9: <unknown function> + 0x7dd5 (0x7efdabb6add5 in /lib64/libpthread.so.0)
frame #10: clone + 0x6d (0x7efdab893ead in /lib64/libc.so.6)
```

This occurs even when the library is built with debugging symbols. The reason this occurs is the symbols in question live in an *anonymous* namespace, and thus don't actually have symbols for `backtrace_symbols` to extract:

```
$ gdb /data/users/ezyang/pytorch-tmp/torch/lib/libtorch_python.so
(gdb) info symbol 0x6a7ee9
c10d::(anonymous namespace)::AsyncSparseAllreduceWork::run() + 41 in section .text  
```

To work around this, you can load the library in question and then ask about the address there (I think we get the symbol name from DWARF in that case). But it would be nice if this just worked out of the box.

**Comment by Avashist1998** [comment]
I am experiencing a similar issue when running the NVIDIA SPADE code base through the training phase. 

System Information
18.04.1-Ubuntu SMP
GP102 [GeForce GTX 1080 Ti]
 NVIDIA-SMI 440.100      Driver Version: 440.100      CUDA Version: 10.2 
Python3 = 3.6.9

Pip Env Information
Package              Version
-------------------- ---------
absl-py              0.9.0
astor                0.8.1
certifi              2020.6.20
chardet              3.0.4
cycler               0.10.0
Cython               0.29.21
decorator            4.4.2
dill                 0.3.2
dominate             2.5.1
filelock             3.0.12
future               0.18.2
gast                 0.2.2
gdown                3.12.0
google-pasta         0.2.0
grpcio               1.30.0
h5py                 2.10.0
idna                 2.10
imageio              2.9.0
importlib-metadata   1.7.0
Keras-Applications   1.0.8
Keras-Preprocessing  1.1.2
kiwisolver           1.2.0
Markdown             3.2.2
matplotlib           3.3.0
networkx             2.4
numpy                1.19.1
opt-einsum           3.3.0
Pillow               7.2.0
pip                  20.2.1
pkg-resources        0.0.0
protobuf             3.12.4
pycocotools          2.0.1
pyparsing            2.4.7
PySocks              1.7.1
python-dateutil      2.8.1
PyWavelets           1.1.1
requests             2.24.0
scikit-image         0.17.2
scipy                1.2.0
setuptools           49.2.1
six                  1.15.0
tensorboard          1.15.0
tensorflow           1.15.0
tensorflow-estimator 1.15.1
termcolor            1.1.0
tifffile             2020.7.24
torch                1.6.0
torchvision          0.7.0
tqdm                 4.48.2
urllib3              1.25.10
Werkzeug             1.0.1
wheel                0.34.2
wrapt                1.12.1
zipp                 3.1.0


 Bash Setup script to download and set up the environment  

#!/bin/bash
# this script is used to setupt the SPADE dataset and the enviornment

git clone https://github.com/NVlabs/SPADE.git
cd SPADE/
source pytorch/bin/activate
pip install -r requirements.txt
cd models/networks/
git clone https://github.com/vacancy/Synchronized-BatchNorm-PyTorch
cp Synchronized-BatchNorm-PyTorch/sync_batchnorm . -rf

cd ../../

# setup of the checkpoint file 
mkdir checkpoints
cd checkpoints
gdown https://drive.google.com/uc?id=12gvlTbMvUcJewQlSEaZdeb2CdOB-b8kQ
tar xvf checkpoints.tar.gz
cd ..

# setting up the coco statset for testing and training
pip install pycocotools

mkdir downloadedData
cd downloadedData

wget http://images.cocodataset.org/zips/train2017.zip
wget http://images.cocodataset.org/zips/val2017.zip
wget http://calvin.inf.ed.ac.uk/wp-content/uploads/data/cocostuffdataset/stuffthingmaps_trainval2017.zip
wget http://images.cocodataset.org/annotations/annotations_trainval2017.zip

unzip -q train2017.zip;  unzip -q val2017.zip
mkdir ../SPADE/datasets/fullcoco_stuff

unzip stuffthingmaps_trainval2017.zip -d ../datasets/fullcoco_stuff
cd ../datasets/fullcoco_stuff
mkdir train_inst; mkdir val_inst
mv train2017 train_label
mv val2017 val_label

cp -a train_label/. train_inst/
cp -a val_label/. val_inst/
cd ../../downloadedData
mv train2017 train_img
mv val2017 val_img
mv val_img/ ../datasets/fullcoco_stuff/
mv train_img/ ../datasets/fullcoco_stuff/
cd ..

CUDA_LAUNCH_BLOCKING=1 python train.py --name train1 --dataset_mode coco --dataroot ~/SPADE/fullcoco_stuff/ --tf_log --gpu_ids 0

**Comment by guozix** [comment]
Got same problem……
